### PR TITLE
Set Access-Control-Allow-Origin to "*" if request has no Origin

### PR DESCRIPTION
**Pull request by tedjp**
Previously it would include the header:
Access-Control-Allow-Origin: undefined

### CODE REVIEW

File: lib/add-cors-headers.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ module.exports = function addCorsHeaders (request, reply) {
     }
   }
 
-  response.headers['access-control-allow-origin'] = request.headers.origin
+  response.headers['access-control-allow-origin'] = request.headers.origin || "*"
```
Comment (gr2m) [inline lib/add-cors-headers.js]: problem is that "*" is not allowed when `access-control-allow-credentials` is set to true

**Comment by gr2m** [comment]
in what case would a request have no origin? I'd say the correct response would be an error

**Comment by tedjp** [comment]
Any request for content that isn't a cross-origin request. For example static images or files on the same server referenced by an HTML page.

Or in my case a simple `curl` or `HEAD` request to verify the headers, which say "undefined" unless I explicitly add an Origin request header.

**Comment by gr2m** [comment]
In that case, it would make sense to leave out the CORS headers entirely, right? We could log a notice for it

**Comment by gr2m** [comment]
This should be resolved in https://github.com/gr2m/CORS-Proxy/releases/tag/v1.3.0

**Pull request closed without merging**